src/SensioLabs/Melody/WorkingDirectory/WorkingDirectoryFactory.php

```diff
@@ -23,19 +23,25 @@ public function __construct($storagePath)
         $this->filesystem->mkdir($this->storagePath);
     }
 
-    public function createTmpDir(array $packages)
+    public function createTmpDir(array $packages, array $repositories)
     {
-        $hash = $this->generateHash($packages);
+        $hash = $this->generateHash($packages, $repositories);
 
         $path = sprintf('%s/%s', $this->storagePath, $hash);
 
         return new WorkingDirectory($path, $this->filesystem);
     }
 
-    private function generateHash(array $packages)
+    private function generateHash(array $packages, array $repositories)
     {
         ksort($packages);
 
-        return hash('sha256', serialize($packages));
+        if (empty($repositories)) {
+            $config = $packages;
+        } else {
+            $config = array($repositories, $packages);
```

Comment:
Packages are sorted at line 37, you should do the same here.
And because repositories are a complete type, you have to write your own sort method (or replace sort a serialization of the repositories like 
```
$repositorie = array_map('serialize', $repositories);
sort($repositories)
```
)